src/ngraph/op/broadcast.cpp

```diff
@@ -15,29 +15,278 @@
 //*****************************************************************************
 
 #include "ngraph/op/broadcast.hpp"
+#include "ngraph/op/constant.hpp"
 #include "ngraph/op/sum.hpp"
+#include "ngraph/op/util/broadcasting.hpp"
+#include "ngraph/partial_shape.hpp"
+
+#include <numeric>
 
 using namespace std;
 using namespace ngraph;
 
-constexpr NodeTypeInfo op::Broadcast::type_info;
+constexpr NodeTypeInfo op::v1::Broadcast::type_info;
+
+op::v1::Broadcast::Broadcast(const Output<Node>& arg,
+                             const Output<Node>& target_shape,
+                             const Output<Node>& axes_mapping,
+                             const AutoBroadcastSpec& broadcast_spec)
+    : Op({arg, target_shape, axes_mapping})
+    , m_broadcast_spec(broadcast_spec)
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v1::Broadcast::Broadcast(const Output<Node>& arg,
+                             const Output<Node>& target_shape,
+                             const AutoBroadcastSpec& broadcast_spec)
+    : Op({arg, target_shape, op::Constant::create(element::u8, Shape{}, {0})->output(0)})
```

Comment:
Also I realized that in case with 2 inputs broadcast_spec must be NUMPY. So let's move it from constructor to set m_broadcast_spec(NUMPY) directly.